WooCommerce/Classes/ViewRelated/Dashboard/Stats v4/StoreStatsPeriodViewModel.swift

```diff
@@ -0,0 +1,309 @@
+        return StatsTimeRangeBarViewModel(startDate: startDate,
+                                          endDate: endDate,
+                                          selectedDate: date,
+                                          timeRange: timeRange,
+                                          timezone: siteTimezone)
+    }
+
+    func createOrderStatsText(orderStatsData: OrderStatsData, selectedIntervalIndex: Int?) -> String {
+        if let count = orderCount(at: selectedIntervalIndex, orderStats: orderStatsData.stats, orderStatsIntervals: orderStatsData.intervals) {
+            return Double(count).humanReadableString()
+        } else {
+            return Constants.placeholderText
+        }
+    }
+
+    func createRevenueStats(orderStatsData: OrderStatsData, selectedIntervalIndex: Int?) -> String {
+        if let revenue = revenue(at: selectedIntervalIndex, orderStats: orderStatsData.stats, orderStatsIntervals: orderStatsData.intervals) {
+            return currencyFormatter.formatHumanReadableAmount(String("\(revenue)"), with: currencyCode) ?? String()
+        } else {
+            return Constants.placeholderText
+        }
+    }
+
+    func createVisitorStatsText(siteStats: SiteVisitStats?, selectedIntervalIndex: Int?) -> String {
+        if let visitorCount = visitorCount(at: selectedIntervalIndex, siteStats: siteStats) {
+            return Double(visitorCount).humanReadableString()
+        } else {
+            return Constants.placeholderText
+        }
+    }
+
+    func createConversionStats(orderStatsData: OrderStatsData, siteStats: SiteVisitStats?, selectedIntervalIndex: Int?) -> String {
+        let visitors = visitorCount(at: selectedIntervalIndex, siteStats: siteStats)
+        let orders = orderCount(at: selectedIntervalIndex, orderStats: orderStatsData.stats, orderStatsIntervals: orderStatsData.intervals)
+        if let visitors = visitors, let orders = orders, visitors > 0 {
+            // Maximum conversion rate is 100%.
+            let conversionRate = min(orders/visitors, 1)
+            let numberFormatter = NumberFormatter()
+            numberFormatter.numberStyle = .percent
+            numberFormatter.minimumFractionDigits = 1
+            return numberFormatter.string(from: conversionRate as NSNumber) ?? Constants.placeholderText
+        } else {
+            return Constants.placeholderText
+        }
+    }
+}
+
+// MARK: - Private data helpers
+//
+private extension StoreStatsPeriodViewModel {
+    func visitorCount(at selectedIndex: Int?, siteStats: SiteVisitStats?) -> Double? {
+        let siteStatsItems = siteStats?.items?.sorted(by: { (lhs, rhs) -> Bool in
+            return lhs.period < rhs.period
+        }) ?? []
+        if let selectedIndex = selectedIndex, selectedIndex < siteStatsItems.count {
+            guard selectedIndex < siteStatsItems.count else {
+                return nil
+            }
```

Comment:
I think the `guard` check here is redundant because the condition has been added in the external `if` check. Should we remove it?

Reply:
Ohh great catch! Removed in https://github.com/woocommerce/woocommerce-ios/pull/5788/commits/3285fc119c0a0dec5459faaae330a9032a743aa5